src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3933,7 +3933,7 @@ void Arguments::PropertyList_unique_add(SystemProperty** plist, const char* k, c
 // expanded one.
 // 2. The passed in "buflen" should be large enough to hold the null terminator.
 bool Arguments::copy_expand_pid(const char* src, size_t srclen,
-                                char* buf, size_t buflen) {
+                                char* buf, size_t buflen, int pid) {
```

Comment:
What does this change do? Why is it needed?

Reply:
It allows us to re-use existing hotspot code by accepting pid into the resulting string, which we need for custom log file names with pid.